src/modules/cpu/kernel/transpose_generic.hpp

```diff
@@ -0,0 +1,441 @@
+        }
+        else
+        {
+            for(int i = 1; i < nDim; i++)
+                srcPtrTemp += begin[i] * srcGenericDescPtr->strides[i];
+
+            if (nDim == 2 && perm[0] == 1 && perm[1] == 0)
+            {
+                compute_2d_transpose(srcPtrTemp, dstPtrTemp, length[0], length[1], srcGenericDescPtr->strides[1], dstGenericDescPtr->strides[1]);
+            }
+            else if (nDim == 3)
+            {
+                if(perm[0] == 2 && perm[1] == 0 && perm[2] == 1 && length[2] == 16)
+                {
+                    Rpp32u height = length[0];
+                    Rpp32u width = length[1];
+                    Rpp32u channels = 16;
+                    Rpp32u bufferLength = width * channels;
+                    Rpp32u alignedLength = (bufferLength / 64) * 64;
+                    Rpp32u vectorIncrement = 64;
+                    Rpp32u vectorIncrementPerChannel = 4;
+
+                    // initialize pointers for 16 channel
+                    Rpp32f *dstPtrChannel[16];
+                    for(int i = 0; i < 16; i++)
+                        dstPtrChannel[i] = dstPtrTemp + i * dstGenericDescPtr->strides[1];
+
+                    // loop over rows
+                    for(int i = 0; i < height; i++)
+                    {
+                        Rpp32f *srcPtrRow = srcPtrTemp;
+
+                        // update temporary pointers for 16 channel
+                        Rpp32f *dstPtrTempChannel[16];
+                        for(int k = 0; k < 16; k++)
+                            dstPtrTempChannel[k] = dstPtrChannel[k];
+
+                        Rpp32u vectorLoopCount = 0;
+                        for( ; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrement)
+                        {
+                            __m256 pSrc[8];
+                            // load 64 values for source
+                            rpp_simd_load(rpp_load8_f32_to_f32_avx, srcPtrRow, &pSrc[0]);
+                            rpp_simd_load(rpp_load8_f32_to_f32_avx, srcPtrRow + 16, &pSrc[1]);
+                            rpp_simd_load(rpp_load8_f32_to_f32_avx, srcPtrRow + 32, &pSrc[2]);
+                            rpp_simd_load(rpp_load8_f32_to_f32_avx, srcPtrRow + 48, &pSrc[3]);
+                            rpp_simd_load(rpp_load8_f32_to_f32_avx, srcPtrRow + 8, &pSrc[4]);
+                            rpp_simd_load(rpp_load8_f32_to_f32_avx, srcPtrRow + 24, &pSrc[5]);
+                            rpp_simd_load(rpp_load8_f32_to_f32_avx, srcPtrRow + 40, &pSrc[6]);
+                            rpp_simd_load(rpp_load8_f32_to_f32_avx, srcPtrRow + 56, &pSrc[7]);
+
+                            __m128 pDst[16];
+                            compute_transpose4x8_avx(&pSrc[0], &pDst[0]);
+                            compute_transpose4x8_avx(&pSrc[4], &pDst[8]);
+
+                            // store 4 values per in output per channel
```

Comment:
Something missing in the language of the comment

Reply:
updated